packages/server/src/user.rs

```diff
@@ -21,42 +21,24 @@ pub struct UserDetails {
 }
 
 impl UserDetails {
-    pub fn get_mut_course_status(&mut self, id: &str) -> Option<&mut CourseStatus> {
-        for course_status in &mut self.degree_status.course_statuses.iter_mut().rev() {
+    pub fn get_course_status(&mut self, id: &str) -> Option<&CourseStatus> {
+        // returns the first course_status with the given id
+        for course_status in &mut self.degree_status.course_statuses.iter() {
```

Comment:
if this changes from mut course status to not mut, it shouldn't take mut self ref.

```rust
...
pub fn get_course_status(&self, id: &str) -> Option<&CourseStatus> {
    ...
    for course_status in self.degree_status.course_statuses.iter() {
    ...
```